packages/datetime/src/dateInput.tsx

```diff
@@ -447,10 +464,37 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
             // the page. tabbing forward should *not* close the popover, because
             // focus will be moving into the popover itself.
             this.setState({ isOpen: false });
+        } else if (e.which === Keys.ESCAPE) {
+            this.setState({ isOpen: false });
+            this.inputRef.blur();
         }
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
+    private registerPopoverBlurHandler = () => {
+        const node = ReactDOM.findDOMNode(this);
+        const tabbableElements = node.querySelectorAll("[tabindex]:not([tabindex='-1'])");
+        const numOfElements = tabbableElements.length;
+        if (numOfElements) {
+            const lastPopoverElement = tabbableElements[numOfElements - 1] as HTMLElement;
+            if (this.lastPopoverElement !== lastPopoverElement) {
+                this.unregisterPopoverBlurHandler();
+                this.lastPopoverElement = lastPopoverElement;
+                this.lastPopoverElement.addEventListener("blur", this.handlePopoverBlur);
+            }
+        }
+    };
+
+    private unregisterPopoverBlurHandler = () => {
+        if (this.lastPopoverElement) {
+            this.lastPopoverElement.removeEventListener("blur", this.handlePopoverBlur);
+        }
+    };
+
+    private handlePopoverBlur = () => {
```

Comment:
no need for this method with optional arg in `handleClosePopover`, just use `handleClosePopover` directly

Reply:
- [ ] remove and call `handleClosePopover`

Reply:
I had to do something like: `(this.handleClosePopover as any) as EventListener` which doesn't look great. Is removing the extra method still more preferable?